packages/playground/blueprints/src/lib/resources.ts

```diff
@@ -212,15 +212,29 @@ export abstract class FetchResource extends Resource {
 	async resolve() {
 		this.progress?.setCaption(this.caption);
 		const url = this.getURL();
-		let response = await fetch(url);
-		response = await cloneResponseMonitorProgress(
-			response,
-			this.progress?.loadingListener ?? noop
-		);
-		if (response.status !== 200) {
-			throw new Error(`Could not download "${url}"`);
+		try {
+			let response = await fetch(url);
+			if (!response.ok) {
+				throw new Error(`Could not download "${url}"`);
+			}
+			response = await cloneResponseMonitorProgress(
+				response,
+				this.progress?.loadingListener ?? noop
+			);
+			if (response.status !== 200) {
+				throw new Error(`Could not download "${url}"`);
+			}
+			return new File([await response.blob()], this.name);
+		} catch (e) {
+			throw new Error(`
+				Could not download "${url}".
+				Check if the URL is correct and the server is reachable.
+				If it's reachable, the server might be blocking the request.
+				Check the console and network for more information.
+				In case of a CORS error, you can try using a proxy server.
```

Comment:
Can we infer what type of error was that and provide a dedicated message outlining the next steps? For example:

* If it's 404 – say file couldn't be found.
* If it's 403 – say it's probably a private resource.
* If it's a CORS error, provide a paragraph or two of text to explain it's a browser security thing, provide a link that explains what's CORS, say they need to move their file somewhere else, e.g. GitHub / raw.githubusercontent.com, link out to a resource explaining setting up CORS headers on their existing servers.

The idea is a person reading that message will know exactly what to do and won't ask us any follow-up questions. If there are no good external reliable resources we could link to, we could add a doc page to link to.

Reply:
[Fetch returns a type error when a network error occurs without details](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful). This means we can't distinguish between different types of network errors. 

404 is special so it will return a response and our [code will return_Could not download_](https://github.com/WordPress/wordpress-playground/blob/a6a255cc23feabaa18f98435a2dbad007484eeed/packages/playground/blueprints/src/lib/resources.ts#L218).

Reply:
Oh bummer! Well, let's provide a longer writeup, then, to at least explain what to do next, even if that's just "Glance at your console, what do you see? 404? do X. CORS? do Y". It could even be a link to a documentation page.

Reply:
I updated the message, let me know if it's still unclear.

Reply:
Now I understand what you mean by longer writeup :smile:

Reply:
Thank you!